### PR TITLE
Update dependencies graph

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Update dependencies graph

**Motivation and Context**
- Why is this change required? What problem does it solve?

The old one is outdated.

- If it fixes an open issue, please link to the issue here.

graph

### CODE REVIEW

File: docs/onnxruntime_dependencies.dot [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    "protobuf" [ label="Google Protobuf" shape="box"];
+    "onnx" -> "protobuf"
+    "ort_graph" -> "protobuf"    
+    "ort_graph" -> "onnx"
+    "ort_optimizer" -> "onnx"        
+    "ort_framework" [ label="onnxruntime_framework" shape="box"];
+    "ort_framework" -> "ort_graph" 
+    "ort_framework" -> "ort_common" 
+    "ort_framework" -> "onnx" 
+    "ort_cpu_provider" [ label="onnxruntime_cpu_provider\n(kernels)" shape="box"];
```
Comment (faxu) [inline docs/onnxruntime_dependencies.dot:26]: other than cpu and cuda, should any other providers be listed? (i.e. DML or others from intel/nvidia/etc)

File: docs/onnxruntime_dependencies.dot [hunk 2]
```diff
@@ -0,0 +1,88 @@
+digraph "GG" {
```
Comment (faxu) [inline docs/onnxruntime_dependencies.dot:1]: Should also add a link to this page from the main readme to make it more easily discoverable, perhaps under https://github.com/microsoft/onnxruntime#system-requirements ?

Comment (faxu) [inline docs/onnxruntime_dependencies.dot:1]: What about dependencies on OpenMP?

File: docs/onnxruntime_dependencies.dot (hunk 1 above)
Comment (snnn) [reply docs/onnxruntime_dependencies.dot:26]: Yes I can. But the graph will be more complex. 
How many we should list there?
Please give me the names.

File: docs/onnxruntime_dependencies.dot (hunk 2 above)
Comment (snnn) [reply docs/onnxruntime_dependencies.dot:1]: Sure.

Comment (snnn) [reply docs/onnxruntime_dependencies.dot:1]: I didn't list all the dependencies here. For example, eigen isn't in the graph.  Yes I could put them there. But we need to find a balance between comprehensive and readable.

File: docs/onnxruntime_dependencies.dot (hunk 1 above)
Comment (jywu-msft) [reply docs/onnxruntime_dependencies.dot:26]: I think it's quite readable in its current state.
We don't need to list every possible dependency/optional components.
EP's is a pretty easy concept to understand, and they are more about extensions to the core. so we can leave other ones out of this (especially externally developed ones)

File: docs/onnxruntime_dependencies.dot [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+"grpc" [ label="gRPC" shape="box"];
+"boost" [ label="Boost" shape="box"];
+"onnx2" [ label="onnx" shape="box"];
+"protobuf2" [ label="Google Protobuf" shape="box"];
+"onnx2" -> "protobuf2"
+"grpc" -> "protobuf2"
+
+subgraph cluster_2 {
+    label = "Applications";
+  "onnxruntime_server" [ label="ONNX Runtime Server" shape="box"];
```
Comment (jywu-msft) [inline docs/onnxruntime_dependencies.dot:78]: do we really want to list onnxruntime server , given its current support status?

Comment (snnn) [reply docs/onnxruntime_dependencies.dot:78]: But, it's still in our repo.

**Pull request merged**